core/domain/question_domain.py

```diff
@@ -172,6 +172,46 @@ def create_default_question_state(cls):
         return state_domain.State.create_default_state(
             None, is_initial_state=True)
 
+    @classmethod
+    def _convert_state_v27_dict_to_v28_dict(cls, state_dict):
```

Comment:
I'm a bit surprised by this. Shouldn't we not be repeating code and putting this in a central state_domain location instead?

Reply:
For exploration, we have a method to convert state's' which we use to do in  exp_domain. I found that this is the patter we have to add state conversion method for the question.  

We can have a pattern like this:

state_domain:
_convert_state_vx_dict_to_vx+1_dict

exp_domain:
_convert_states_vx_dict_to_vx+1_dict:

for state in state:
  state_domain._convert_state_vx_dict_to_vx+1_dict
......

in question_domain:

directly use the state_domain conversion methods.

(I'm not sure about this structure.) what's your thought?

Reply:
I think we should find a way that does not have duplicate code. Maybe exp_domain's "upgrade exp" and question_domain's "upgrade question" should call conversion methods in state_domain and all the existing state conversion methods should be moved to state_domain.